tests/conversion_02.cc

```diff
@@ -0,0 +1,68 @@
+
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the SampleFlow authors.
+//
+// This file is part of the SampleFlow library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Check the conversion filter
+
+
+#include <iostream>
+
+#include <sampleflow/consumers/stream_output.h>
+#include <sampleflow/filters/conversion.h>
+#include <sampleflow/producers/range.h>
+
+/**
+ * Example conversion function; converts a string to an integer
+ * by getting the sum of the ASCII codes of each character.
+ */
+int string_ascii_sum(std::string sample)
```

Comment:
General comment: You're passing the argument by value here, which requires a copy. But you don't intend to actually do anything that requires a copy, you could as well work on the original value. In that case, for anything other than scalar types (int, double, etc), the way to do that is to take the argument by reference, and because you're not going to change it, by `const` reference. So that would be this:
```suggestion
int string_ascii_sum(const std::string &sample)
```